program/rust/src/rust_oracle.rs

```diff
@@ -22,6 +96,42 @@ pub fn update_version(
     accounts: &Vec<AccountInfo>,
     instruction_data: &[u8],
 ) -> OracleResult {
-    panic!("Need to merge fix to pythd in order to implement this");
+    //read account info
+    let pyth_acc_info_size = size_of::<pc_acc_t>();
+    let mut pyth_acc_info =
+        pc_acc_t::try_from_slice(&accounts[1].try_borrow_data()?[..pyth_acc_info_size])?;
+
+    //check magic number
+    if pyth_acc_info.magic_ != PC_MAGIC {
+        //TODO: once we settle on error stuff fix this
+        return Err(OracleError::IncorrectMagicNumber.into());
+    }
```

Comment:
Can we have a function that checks that the account is a valid oracle account? It might return the type of the account.

Reply:
It needs to check the owner of the account + magic number and other things that we might need to check.

Reply:
Why check for the owner? you should not be able to modify an account that you do not own?
But yes sure, I guess it won't harm